### PR TITLE
Add docs for KEP-4017 (pod index label)

### PR DESCRIPTION
**Pull request by danielvegamyhre**
The purpose of these changes are to add documentation for the new pod index label being added for StatefulSets and Indexed Jobs (see [KEP-4017](https://github.com/kubernetes/enhancements/pull/4019)).

### CODE REVIEW

**Comment by danielvegamyhre** [comment]
tagging @sftim @alculquicondor for review. I created a new branch from dev-1.28 and copied over the approved changes from https://github.com/kubernetes/website/pull/42059 here, since in the original PR I accidentally branched off of main which was causing issues when setting dev-1.28 as the merge target.

File: content/en/docs/concepts/workloads/controllers/job.md [hunk 1]
```diff
@@ -278,8 +278,11 @@ Jobs with _fixed completion count_ - that is, jobs that have non null
   completion is homologous to each other. Note that Jobs that have null
   `.spec.completions` are implicitly `NonIndexed`.
 - `Indexed`: the Pods of a Job get an associated completion index from 0 to
-  `.spec.completions-1`. The index is available through three mechanisms:
+  `.spec.completions-1`. The index is available through four mechanisms:
   - The Pod annotation `batch.kubernetes.io/job-completion-index`.
+  - The Pod label `batch.kubernetes.io/job-completion-index` (for v1.28 and later). Note
+  the feature gate `PodIndexLabel` must be enabled to use this label, and it is enabled
+  by default.
```
Comment (windsonsea) [inline content/en/docs/concepts/workloads/controllers/job.md:285]:
```suggestion
  - The Pod label `batch.kubernetes.io/job-completion-index` (for v1.28 and later). Note
    the feature gate `PodIndexLabel` must be enabled to use this label, and it is enabled
    by default.
```
Could it has a consistent style with the context nearby?

**Review by sftim — COMMENTED** [review]
I think this is ready for tech review (by SIG Apps)

**Comment by danielvegamyhre** [comment]
tagging @soltysh for sig apps review

**Comment by alculquicondor** [comment]
LGTM as job approver

/assign @janetkuo 
(soltysh is on vacation)

File: content/en/docs/reference/labels-annotations-taints/_index.md [hunk 2]
```diff
@@ -1033,13 +1047,13 @@ by the cloud-controller-manager.
...
 
-Type: Annotation
+Type: Annotation, Label
 
 Example: `batch.kubernetes.io/job-completion-index: "3"`
 
 Used on: Pod
 
-The Job controller in the kube-controller-manager sets this annotation for Pods
+The Job controller in the kube-controller-manager sets this as a label and annotation for Pods
```
Comment (janetkuo) [inline content/en/docs/reference/labels-annotations-taints/_index.md:1056]: Why do we need the same information as both a label or annotation?

File: content/en/docs/reference/labels-annotations-taints/_index.md [hunk 3]
```diff
@@ -226,6 +226,20 @@ This annotation is applied to the parent object used to track an ApplySet to ind
...
+
+Example: `apps.kubernetes.io/pod-index: "0"`
+
+Used on: Pod
+
+When a StatefulSet controller creates a Pod for the StatefulSet, it sets this label on that Pod. 
+The value of the label is the ordinal index of the pod being created.
+
+See [Pod Index Label](/docs/concepts/workloads/controllers/statefulset/#pod-index-label)
+in the StatefulSet topic for more details.
```
Comment (janetkuo) [inline content/en/docs/reference/labels-annotations-taints/_index.md:241]: Maybe mention feature gate here and below?

File: content/en/docs/reference/labels-annotations-taints/_index.md (hunk 2 above)
Comment (danielvegamyhre) [reply content/en/docs/reference/labels-annotations-taints/_index.md:1056]: See [KEP-4017](https://github.com/kubernetes/enhancements/pull/4019) for rationale behind this decision.

File: content/en/docs/reference/labels-annotations-taints/_index.md (hunk 3 above)
Comment (danielvegamyhre) [reply content/en/docs/reference/labels-annotations-taints/_index.md]: Added reference to feature gate explaining it must be enabled for this pod label to be added.

**Comment by danielvegamyhre** [comment]
@janetkuo I addressed your comments and this is ready for another look

**Review by janetkuo — APPROVED** [review]
/lgtm

**Comment by danielvegamyhre** [comment]
@sftim for approval

**Review by sftim — COMMENTED** [review]
Thanks

/approve

**Pull request merged**